File: JS/1957.delete_characters_to_make_fancy_string [1].js
/**
 * @param {string} s
 * @return {string}
 */
var makeFancyString = function(s) {
        let len = 1
        let ans = ''
        ans += s[0]

        for(let i=1; i<s.length; i++){
            if(s[i-1] === s[i])
                len++
            else
                len = 1
                
            if(len < 3)
                ans += s[i]
        }

        return ans
};

File: CPP [1501-2000]/1957.delete-characters-to-make-fancy-string [1].cpp
class Solution {
public:
    // T.C.=O(n), S.C.=O(1)
    string makeFancyString(string s) {
        int len = 1;
        string ans = "";
        ans += s[0];

        for(int i=1; i<s.length(); i++)
        {
            if(s[i-1] == s[i])
                len++;
            else
                len = 1;
                
            if(len < 3)
                ans += s[i];
        }

        return ans;
    }
};